packages/core/src/domain/user/user.ts

```diff
@@ -0,0 +1,18 @@
+import type { Context } from '../../tools/context'
+import { deepClone } from '../../tools/utils'
+
+/**
+ * Clone input data and ensure known user properties (id, name, email)
+ * are strings, as defined here:
+ * https://docs.datadoghq.com/logs/log_configuration/attributes_naming_convention/#user-related-attributes
+ */
+export function sanitizeUser(newUser: Context): Context {
+  const user = deepClone(newUser)
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏any reason to deepClone here instead of shallow clone?

Reply:
It's possible to receive nested objects. It should not happen often, but it's a precaution.

Reply:
Example:
```
const obj = { foo: 'bar' }
const nested = { biz: 'buz' };
obj.nested = nested

console.log(obj) // { foo: 'bar', nested: { biz: 'buz' } }

nested.biz = 'break it'

console.log(obj) // { foo: 'bar', nested: { biz: 'break it' } }
```

Reply:
Yes, since the Context Manager [already deep clones it when getting/setting the user](https://github.com/DataDog/browser-sdk/blob/main/packages/core/src/tools/contextManager.ts#L27-L32), this seems unnecessary. See @bcaudan remark here: https://github.com/DataDog/browser-sdk/pull/1641#discussion_r932438621